applications/maplab-server-node/src/maplab-server-ros-node.cc

```diff
@@ -134,26 +148,41 @@ bool MaplabServerRosNode::start() {
   LOG(INFO) << "[MaplabServerRosNode] Starting...";
   // Start the pipeline. The ROS spinner will handle SIGINT for us and abort
   // the application on CTRL+C.
+  bool restore_server_state;
+  nh_.param<bool>(
+      FLAGS_maplab_server_restore_previous_state_param_name,
+      restore_server_state, false);
+  maplab_server_node_->start(restore_server_state);
   maplab_spinner_.start();
-  maplab_server_node_->start();
+  if (restore_server_state) {
+    ros::ServiceClient query_all_subfolders_client =
+        nh_.serviceClient<transfolder_msgs::QueryAllSubfolders>(
+            FLAGS_maplab_server_transfolder_query_all_subfolders_service, true);
+    if (query_all_subfolders_client.waitForExistence(ros::Duration(1))) {
+      transfolder_msgs::QueryAllSubfolders subfolder_msg;
+      query_all_subfolders_client.call(subfolder_msg);
```

Comment:
this call will trigger transfolder to publish all available submaps, so we can compare which ones were already included in the restored state and which ones still have to be loaded